Backend/dev/config-sync/config_transfer.py

```
@@ -489,25 +532,92 @@ def list_table_files(base_dir, env_name, schema):
+
+def fetch_patch_zip_from_cloudfront(direction: str, cloudfront_url: str, dest_dir: Path):
+    """Download <cloudfront_url>/<direction>.zip and unzip into dest_dir."""
+    import tempfile
+    import urllib.error
+    import urllib.request
+    import zipfile
+
+    zip_url = f"{cloudfront_url.rstrip('/')}/{direction}.zip"
+    print(f"  [fetch] {zip_url}")
+    try:
+        with urllib.request.urlopen(zip_url, timeout=120) as resp:
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)